### PR TITLE
Update jwt when user profile changes

### LINKED ISSUE #132: update user jwt on profile change

### PR DESCRIPTION
**Pull request by hahn-kev**
This contributes to #126, and it closes #132

### CODE REVIEW

File: frontend/src/routes/(authenticated)/myaccount/+page.ts [hunk 1]
```diff
@@ -32,6 +33,9 @@ export async function _changeUserAccountData(input: ChangeUserAccountDataInput):
       //invalidates the graphql user cache, but who knows
       { additionalTypenames: ['Users'] },
     );
-  if (!result.error) void invalidate(`user:${input.userId}`);
+  if (!result.error) {
+    await refreshJwt();
+    await invalidate(`user:${input.userId}`);
```
Comment (rmunn) [inline frontend/src/routes/(authenticated)/myaccount/+page.ts:38]: I believe we'll be able to get rid of the `{#if changed}` stanza at the bottom of `+page.svelte`, too:

```
{#if changed}
    <p class="text-warning">{$t('account_settings.notify')}</p>
{/if}
```

The invalidate re-runs load, which should automatically refresh the values on the page, and there should be no need to tell the user they need to sign out and sign back in.

File: backend/LexBoxApi/Controllers/LoginController.cs [hunk 2]
```diff
@@ -49,20 +49,25 @@ public async Task<ActionResult<LexAuthUser>> Login(LoginRequest loginRequest)
         return user;
     }
 
+    [HttpPost("refresh")]
```
Comment (rmunn) [inline backend/LexBoxApi/Controllers/LoginController.cs:52]: A POST with no content seems a bit weird, but it's possible that in the future we'll decide to add content here, so let's leave this as a POST.

**Review by rmunn — APPROVED** [review]
LGTM. I haven't had time to test this yet, but the design looks right.

File: frontend/src/routes/(authenticated)/myaccount/+page.ts (hunk 1 above)
Comment (dadukhankevin) [reply frontend/src/routes/(authenticated)/myaccount/+page.ts:38]: That will be gone when i submit a new pr

**Review by myieye — COMMENTED** [review]
I rebased it and fixed the conflicts just for fun.
I'm running this and not seeing my token change. Is it just me or is something missing here?

File: backend/LexBoxApi/Auth/LexAuthService.cs [hunk 3]
```diff
@@ -71,6 +71,13 @@ public async Task ForgotPassword(string email)
         return validPassword ? lexAuthUser : null;
     }
 
+    public async Task<LexAuthUser?> RefreshUser(Guid userId)
```
[the remark this answers is not in the document]
Comment (hahn-kev) [reply backend/LexBoxApi/Auth/LexAuthService.cs:74]: I considered that but this method would return null if the user account was locked or soft deleted, so I didn't want it to just be a get method. But I agree the name isn't great.

**Comment by hahn-kev** [comment]
Thanks. Uh I'm not sure, does it return a set cookie header? It's possible if nothing has changed dotnet won't change the cookie but that seems unlikely.

**Comment by myieye** [comment]
> Thanks. Uh I'm not sure, does it return a set cookie header? It's possible if nothing has changed dotnet won't change the cookie but that seems unlikely.

No, no set cookie header.

**Pull request merged**